src/controllers/contract.ts

```diff
@@ -38,19 +38,17 @@ const syncDataToSmartContract = async (_req: Request, res: Response) => {
 
   // Submit a batch of data to the smart contract to sync
   try {
-    /* TODO: uncomment this
     let tx = await contract.syncState(users, tags, urls, likes);
     await tx.wait();
-    */
   } catch (error: any) {
     console.error("error: ", error);
     return res.status(500).json({ error: `Failed to sync state: ${error.message}` });
   }
 
   // Mark all as synced
-  await UserControl.markSynced(users);
-  await TagControl.markSynced(tags);
-  await URLControl.markSynced(urls);
+  await UserControl.markSynced(users.map(user => user.userAddress as string));
```

Comment:
Nico can you explain why this change? instead of complete users , tag and url object we are taking in only name, url and address? On smart contract side I assume only these 3 are needed right?

Reply:
The `getUsersToSync` is returning in the format/type that the smart contract accepts. We get this type from the typechain generated (you need to run `npm run typechain` to run it. I added the instruction in the readme).

I decided that `markSync` functions should not use the typechain to avoid using a smart contract type in a function that does not interact with the contract. That is why the `markSync` functions usually use a string id rather than the object itself (user address, url, tag name, etc)